pkg/manager/member/executor_manager.go

```diff
@@ -725,7 +718,158 @@ func (m *executorMemberManager) getContainerByName(sts *appsv1.StatefulSet, cont
 	return nil
 }
 
-func (m *executorMemberManager) getVolsByName(tc *v1alpha1.TiflowCluster, sts *appsv1.StatefulSet, podName string) (map[string]*v1alpha1.StorageVolumeStatus, error) {
+func (m *executorMemberManager) syncVolsStatus(tc *v1alpha1.TiflowCluster, sts *appsv1.StatefulSet) error {
 	// todo:
-	return nil, nil
+	selector, err := metav1.LabelSelectorAsSelector(sts.Spec.Selector)
+	if err != nil {
+		klog.Errorf("converting statefulSet %s selector to metav1 selector", sts.Name)
+		return err
+	}
+
+	pvs, err := m.clientSet.CoreV1().PersistentVolumes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+
+	for _, pv := range pvs.Items {
+		tc.Status.Executor.Volumes[pv.Name] = &v1alpha1.StorageVolumeStatus{
+			Name: DefaultStorageName,
+			ObservedStorageVolumeStatus: v1alpha1.ObservedStorageVolumeStatus{
+				BoundCount:      1,
+				CurrentCount:    1,
+				ResizedCount:    1,
+				CurrentCapacity: resource.Quantity{},
+				ResizedCapacity: resource.Quantity{},
+			},
+		}
+	}
+
+	return nil
+}
+
+func (m *executorMemberManager) syncExecutorMembersStatus(tc *v1alpha1.TiflowCluster) (map[string]v1alpha1.ExecutorMember, error) {
+	if tc.Heterogeneous() && tc.Spec.Master == nil {
+		return m.syncExecutorsStatusWithHetero(tc)
+	}
+
+	return m.syncExecutorsStatusWithoutHetero(tc)
+}
+
+func (m *executorMemberManager) syncExecutorsStatusWithoutHetero(tc *v1alpha1.TiflowCluster) (map[string]v1alpha1.ExecutorMember, error) {
```

Comment:
Can we combine this function with `syncExecutorsStatusWithHetero`? They look almost the same

Reply:
I'll try